source/frontend/src/features/properties/map/MapRouter.tsx

```diff
@@ -182,6 +183,16 @@ export const MapRouter: React.FunctionComponent<IMapRouterProps> = memo(props =>
         key={'NewProject'}
         title={'Create Project'}
       />
+      <AppRoute
+        path={`/mapview/sidebar/project/:id`}
+        customRender={({ match }) => (
+          <ProjectContainer projectId={Number(match.params.id)} onClose={onClose} />
+        )}
+        claim={Claims.PROJECT_VIEW}
+        exact
+        key={'Project'}
+        title={'Project'}
+      />
```

Comment:
Please use the dependency injection pattern for Containers -> Views
See an example in **AddFinancialCodeContainer**. We are moving to a new pattern where the container gets the inner view injected via props to help with testing and decoupling.

```tsx
          <AppRoute
            protected
            exact
            path="/admin/financial-code/new"
            customRender={() => (
              <AuthLayout>
                <AddFinancialCodeContainer View={AddFinancialCodeForm} />
              </AuthLayout>
            )}
            role={Roles.SYSTEM_ADMINISTRATOR}
            title={getTitle('Create Financial Code')}
          ></AppRoute>
```

Notice how we "inject" the formik form into the container above. This allows us to create simpler forms for testing so the container tests don't have to rely on knowing about the inner children

Reply:
Updated